### ISSUE TITLE
Component: Password input

### ISSUE THREAD

**Issue by asudoh** [carbon-design-system/carbon-web-components#305]

**Comment by oliviaflory** [comment]
Useful for teams with any type of login or password (landing pages, URX)
Would be nice to have feature parody

Seem to have some naming differences between [Carbon React](https://react.carbondesignsystem.com/?path=/story/components-textinput--default) and [Web components](https://web-components.carbondesignsystem.com/?path=/story/components-input--default) storybooks
- `TextInput` vs `Input`

**Comment by emyarod** [comment]
partially blocked by #945
